kaspresso/src/main/kotlin/com/kaspersky/kaspresso/interceptors/behavior/impl/autoscroll/AutoScrollToAction.kt

```diff
@@ -0,0 +1,94 @@
+package com.kaspersky.kaspresso.interceptors.behavior.impl.autoscroll
+
+import android.view.View
+import android.view.ViewParent
+import android.widget.HorizontalScrollView
+import android.widget.ListView
+import android.widget.ScrollView
+import androidx.core.widget.NestedScrollView
+import androidx.test.espresso.PerformException
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.util.HumanReadables
+import com.kaspersky.kaspresso.logger.UiTestLogger
+import io.github.kakaocup.kakao.common.actions.NestedScrollToAction
+
+class AutoScrollToAction(
+    private val logger: UiTestLogger,
+    private val viewAction: ViewAction = NestedScrollToAction(),
+) : ViewAction by viewAction {
+
+    private fun View?.isScrollable(): Boolean =
+        (this is ScrollView || this is NestedScrollView || this is HorizontalScrollView || this is ListView)
+
+    private tailrec fun View?.findFirstParentScrollableView(lastView: View): View? {
+        return when {
+            this.isScrollable() -> this
+            this == lastView -> null
+            else -> this?.parent?.findFirstView()?.findFirstParentScrollableView(lastView)
+        }
+    }
+
+    private tailrec fun ViewParent?.findFirstView(): View? {
+        return if (this is View) {
+            this
+        } else {
+            this?.parent.findFirstView()
+        }
+    }
+
+    override fun perform(uiController: UiController, view: View) {
+        try {
+            viewAction.perform(uiController, view)
+        } catch (exception: PerformException) {
+            fallbackAutoScroll(uiController, view)
+        }
+    }
+
+    private fun fallbackAutoScroll(uiController: UiController, view: View) {
+        val scrollView = view.findFirstParentScrollableView(view.rootView)
+
```

Comment:
I think first we should check the visible area percentage here, as `PerformException` doesn't tell us strictly that there is a need to scroll. It can be any other issue.

Something like:
```
if (ViewMatchers.isDisplayingAtLeast(VISIBLE_AREA_PERCENTAGE).matches(view)) {
    logger.i(TAG, "View is displayed. Returning.")
    return
}
```

Reply:
- [ ] Necessary to check the visible area percentage?

Reply:
Thanks, and I think we can merge then, yes

Reply:
Done! Feel free to review whenever you have time